var/spack/repos/builtin/packages/py-gmxapi/package.py

```diff
@@ -30,9 +31,10 @@ class PyGmxapi(PythonPackage):
     depends_on("py-numpy@1.8:", type=("build", "run"))
     depends_on("py-setuptools@42:", type="build")
     depends_on("py-packaging", type=("build", "run"))
-    depends_on("py-pybind11@2.6:", type=("build", "run"))
+    depends_on("py-pybind11@2.6:", type="build")
+    depends_on("py-pybind11@2.6:", when="@:0.4", type=("build", "run"))
     depends_on("py-pytest@4.6:", type="test")
-    depends_on("py-wheel", type="build")
+    depends_on("py-wheel", when="@:0.4", type="build")
```

Comment:
wheel is automatically added by the base class so you could remove this

Reply:
Thanks!

Reply:
Are you saying that the module files add all deps to the PYTHONPATH, but also recursively load all other modules, which add them to the PYTHONPATH several more times? I'm not very familiar with our modules support, that's @alalazo's territory.

Reply:
Regardless, I don't think there's anything wrong with the package, it would likely be something wrong in the Python recipe or in our module file generation. So I think it's safe to merge this PR.